bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/widgets/TypedListener.java

```diff
@@ -31,13 +32,16 @@
  *
  * @see Listener
  * @see <a href="http://www.eclipse.org/swt/">Sample code and further information</a>
+ * @noreference This class is not intended to be referenced by clients.
+ * @noextend This class is not intended to be subclassed by clients.
+ * @noinstantiate This class is not intended to be instantiated by clients.
```

Comment:
I wonder if we should move this class into the `o.e.swt.internal` package if it is meant to be internal?
As far as I can tell technically it should be possible.
There are ~5 direct references in the SDK and I'll check if they can be replaced with API.

Does anybody has objections?

Reply:
I won't touch this, even though it is marked as such it is visible to client and possibly break things.

Reply:
Yes, I have objections. In my workspace I see our internal code using that in tests and also in 3rd party libraries like swtchart & cdt. Of course no one reads javadoc saying "don't use me!" on a public class, but moving it now would be not nice...

Reply:
I'm fine with either way. I wouldn't have a problem with moving, but I also have no strong preference to do it, so I'll keep it as it is if that's your preference.
I will just add the api-filters to suppress the error due to adding the javadoc-tags (without that API-tools complains that a protected field of a API class has changed its type).

Reply:
Could we merge this **after** merging API checks enablement in SWT?
In the IDE we can only see API errors for the current platform.

This should not show any error on any platform.